src/tabset/tabset.ts

```diff
@@ -19,27 +21,43 @@ export class NgbTabContent {
   constructor(public templateRef: TemplateRef<any>) {}
 }
 
+/**
+ * A directive representing an individual tab.
+ */
 @Directive({selector: 'ngb-tab'})
 export class NgbTab {
+  /**
+   * Unique tab identifier. Must be unique for the entire document for proper accessibility support.
+   */
+  @Input() id: string = `ngb-tab-${nextId++}`;
```

Comment:
this is why typescript rocks.  we don't have to say what the data type is anymore.